src/utils/fetchData.ts

```diff
@@ -0,0 +1,53 @@
+import { Buffer } from 'buffer';
+
+export function fetchText(uri?: string): Promise<string | null> {
+  if (uri && uri.startsWith('data:')) {
+    return new Promise((resolve) => {
+      resolve(dataUriToXml(uri));
+    });
+  } else {
+    return localFetch(uri);
+  }
```

Comment:
Can we use that flow exclusively on Android since it's the only platform that wasn't working?

Reply:
I included a check for the Android platform and enhanced the validation for the URI path.